test/awaitility.go

```diff
@@ -0,0 +1,30 @@
+package test
+
+import (
+	"context"
+	"time"
+)
+
+type Awaitility struct {
+	timeout time.Duration
+}
+
+func WaitWithTimeout(timeout time.Duration) Awaitility {
+	return Awaitility{timeout: timeout}
+}
+
+func (a Awaitility) Until(do func() error) error {
+	ctx, cancel := context.WithTimeout(context.Background(), a.timeout)
+	defer cancel()
+	var err error
+	for {
+		select {
+		case <-ctx.Done():
+			return err
+		case <-time.After(500 * time.Millisecond):
```

Comment:
Also we are talking to openshift master when this is called, I am not sure will it be changed(some resources inside namespace at every iteration) in every 0.5 second. Maybe increase it `1 or 2 s`

Reply:
@MatousJobanek if an event comes from the `ctx.Done()` channel, then it doesn't really matter how much time you'll wait, does it ? My point is, giving `1s` or `2s` of timeout just gives a little more time for the operation to complete, but it's not going to block the test for the entire `timeout` duration.

Reply:
well, currently I'm running these checks in goroutine, which means that changing it to `1s` doesn't influence the resulting time significantly. Originally it was different - it was executed in for cycle sequentially and if you multiply the number of checks by half a sec, then you will get a quite big time difference. Then, I have one test that runs this wait logic in sequence 12-times - this is the difference of 6 seconds. 
I'm willing to change it to `1s` but not `2s`. Still, my main question is - what is the benefit of changing it?

Reply:
I am not sure if we get some object state change with every `0.5s`.

Reply:
changed to `1s` 6a41e1b